### PR TITLE
Fix error wrap on pki_helpers

### LINKED ISSUE #1210: kubeadm swallows errors when CA key isn't in RSA format
**What happened**: When trying to deploy Kubernetes using kubeadm v1.12.1, kubeadm failed to generate certificates, causing it to fail at a later step.

After passing `-v10` to get all the logs I could I found that it was expecting _all_ the certificates to have been pre-made, even though I only added the certificate and key for the certificate authority. So I dug into the code, and in `pkiutil.TryLoadKeyFromDisk` I found this:

```
	// Allow RSA format only
	var key *rsa.PrivateKey
	switch k := privKey.(type) {
	case *rsa.PrivateKey:
		key = k
	default:
		return nil, fmt.Errorf("the private key file %s isn't in RSA format", privateKeyPath)
	}
```
It turns out all the PKI I've been generating for things like etcd has been ECDSA. So I found the issue I was having, but I still couldn't figure out why kubeadm wasn't failing right away. I took a look at the caller of that function:

```
			// Try and load a CA Key
			caKey, err = pkiutil.TryLoadKeyFromDisk(ic.CertificatesDir, ca.BaseName)
			if err != nil {
				// If there's no CA key, make sure every certificate exists.
				for _, leaf := range leaves {
					cl := certKeyLocation{
						pkiDir:   ic.CertificatesDir,
						baseName: leaf.BaseName,
						uxName:   leaf.Name,
					}
					if err := validateSignedCertWithCA(cl, caCert); err != nil {
						return fmt.Errorf("could not load expected certificate %q or validate the existence of key %q for it: %v", leaf.Name, ca.Name, err)
					}
				}
				// CACert exists and all clients exist, continue to next CA.
				continue
			}
```

It looks like that original error is swallowed.

**What you expected to happen**: Immediate failure when keys are in the wrong format.

**How to reproduce it (as minimally and precisely as possible)**: Use an ECDSA cert/key pair and run `kubeadm init`.

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): v1.12.1
- Cloud provider or hardware configuration: OpenStack
- OS (e.g. from /etc/os-release): Ubuntu 18.04
- Kernel (e.g. `uname -a`): 4.15.0-34-generic #37-Ubuntu SMP Mon Aug 27 15:21:48 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
- Install tools: kubeadm
- Others:

/kind bug

### LINKED ISSUE #70594: kubeadm swallows errors when CA key isn't in RSA format
**What happened**: When trying to deploy Kubernetes using kubeadm v1.12.1, kubeadm failed to generate certificates, causing it to fail at a later step.

After passing `-v10` to get all the logs I could I found that it was expecting _all_ the certificates to have been pre-made, even though I only added the certificate and key for the certificate authority. So I dug into the code, and in `pkiutil.TryLoadKeyFromDisk` I found this:

```
	// Allow RSA format only
	var key *rsa.PrivateKey
	switch k := privKey.(type) {
	case *rsa.PrivateKey:
		key = k
	default:
		return nil, fmt.Errorf("the private key file %s isn't in RSA format", privateKeyPath)
	}
```
It turns out all the PKI I've been generating for things like etcd has been ECDSA. So I found the issue I was having, but I still couldn't figure out why kubeadm wasn't failing right away. I took a look at the caller of that function:

```
			// Try and load a CA Key
			caKey, err = pkiutil.TryLoadKeyFromDisk(ic.CertificatesDir, ca.BaseName)
			if err != nil {
				// If there's no CA key, make sure every certificate exists.
				for _, leaf := range leaves {
					cl := certKeyLocation{
						pkiDir:   ic.CertificatesDir,
						baseName: leaf.BaseName,
						uxName:   leaf.Name,
					}
					if err := validateSignedCertWithCA(cl, caCert); err != nil {
						return fmt.Errorf("could not load expected certificate %q or validate the existence of key %q for it: %v", leaf.Name, ca.Name, err)
					}
				}
				// CACert exists and all clients exist, continue to next CA.
				continue
			}
```

It looks like that original error is swallowed.

**What you expected to happen**: Immediate failure when keys are in the wrong format.

**How to reproduce it (as minimally and precisely as possible)**: Use an ECDSA cert/key pair and run `kubeadm init`.

**Anything else we need to know?**:

**Environment**:
- Kubernetes version (use `kubectl version`): v1.12.1
- Cloud provider or hardware configuration: OpenStack
- OS (e.g. from /etc/os-release): Ubuntu 18.04
- Kernel (e.g. `uname -a`): 4.15.0-34-generic #37-Ubuntu SMP Mon Aug 27 15:21:48 UTC 2018 x86_64 x86_64 x86_64 GNU/Linux
- Install tools: kubeadm
- Others:

/kind bug

### PR DESCRIPTION
**Pull request by yagonobre**
**What type of PR is this?**
/kind bug

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes https://github.com/kubernetes/kubeadm/issues/1210
Closes #70594

```release-note
NONE
```

### CODE REVIEW

File: cmd/kubeadm/app/util/pkiutil/pki_helpers.go [hunk 1]
```diff
@@ -201,7 +201,7 @@ func TryLoadKeyFromDisk(pkiPath, name string) (*rsa.PrivateKey, error) {
 	case *rsa.PrivateKey:
 		key = k
 	default:
-		return nil, errors.Wrapf(err, "the private key file %s isn't in RSA format", privateKeyPath)
+		return nil, errors.Errorf("the private key file %s isn't in RSA format", privateKeyPath)
```
Comment (yagonobre) [inline cmd/kubeadm/app/util/pkiutil/pki_helpers.go]: If you try to wrap a `nil` error, this will result in `nil` instead of a new error

**Comment by fabriziopandini** [comment]
@yagonobre Good catch!
/ok-to-test
/approve
/lgtm

**Comment by neolit123** [comment]
/hold
please see my comment.

File: cmd/kubeadm/app/util/pkiutil/pki_helpers.go (hunk 1 above)
Comment (yagonobre) [reply cmd/kubeadm/app/util/pkiutil/pki_helpers.go]: Good catch, fixed!

**Comment by yagonobre** [comment]
@neolit123 @fabriziopandini updated.

**Pull request merged**